Cargo.toml

```diff
@@ -1,27 +1,101 @@
+edition = "2021"
+homepage = "https://www.parity.io/"
+keywords = ["wasm", "parity", "webassembly", "blockchain", "edsl"]
+license = "Apache-2.0"
+repository = "https://github.com/paritytech/ink"
+version = "4.2.0"
+
+[workspace.dependencies]
+arrayref = { version = "0.3" }
+array-init = { version = "2.0", default-features = false }
+blake2 = { version = "0.10" }
+cargo_metadata = { version = "0.17.0" }
+cfg-if = { version = "1.0" }
+contract-build = { version = "3.0.0" }
+derive_more = { version = "0.99.17", default-features = false }
+either = { version = "1.5", default-features = false }
+funty = { version = "2.0.0" }
+heck = { version = "0.4.0" }
+impl-serde = { version = "0.4.0", default-features = false }
+itertools = { version = "0.11", default-features = false }
+jsonrpsee = { version = "0.19.0" }
+linkme = { version = "0.3.9" }
+num-traits = { version = "0.2", default-features = false }
+paste = { version = "1.0" }
+pretty_assertions = { version = "1" }
+proc-macro2 = { version = "1" }
+quickcheck = { version = "1" }
+quickcheck_macros = { version = "1" }
+quote = { version = "1" }
+rlibc = { version = "1" }
+scale = { package = "parity-scale-codec", version = "3.4", default-features = false, features = ["derive"] }
+scale-decode = { version = "0.7.0", default-features = false }
+scale-encode = { version = "0.3.0", default-features = false }
+scale-info = { version = "2.6", default-features = false }
+schemars = { version = "0.8" }
+secp256k1 = { version = "0.27.0" }
+serde = { version = "1.0.137", default-features = false }
+serde_json = { version = "1.0.81" }
+sha2 = { version = "0.10" }
+sha3 = { version = "0.10" }
+static_assertions = { version = "1.1" }
+subxt = { version = "0.29.0" }
+syn = { version = "2" }
+synstructure = { version = "0.13.0" }
+tokio = { version = "1.18.2" }
+tracing = { version = "0.1.37" }
+tracing-subscriber = { version = "0.3.17" }
+trybuild = { version = "1.0.60" }
+which = { version = "4.4.0" }
+xxhash-rust = { version = "0.8" }
+
+# Substrate dependencies
+pallet-contracts-primitives = { version = "24.0.0", default-features = false }
+sp-core = { version = "21.0.0", default-features = false }
+sp-keyring = { version = "24.0.0", default-features = false }
+sp-runtime = { version = "24.0.0", default-features = false }
+sp-weights = { version = "20.0.0", default-features = false }
+
+# Local dependencies
+ink = { version = "=4.2.0", path = "crates/ink", default-features = false }
+ink_allocator = { version = "=4.2.0", path = "crates/allocator", default-features = false }
+ink_codegen = { version = "=4.2.0", path = "crates/ink/codegen", default-features = false }
+ink_e2e_macro = { version = "=4.2.0", path = "crates/e2e/macro", default-features = false }
+ink_engine = { version = "=4.2.0", path = "crates/engine", default-features = false }
+ink_env = { version = "4.2.0", path = "crates/env", default-features = false }
+ink_ir = { version = "4.2.0", path = "crates/ink/ir", default-features = false }
```

Comment:
Why are these two not fixed?

Reply:
thanks for spotting: https://github.com/paritytech/ink/pull/1856